src/main/java/com/javadeobfuscator/deobfuscator/transformers/stringer/StringEncryptionTransformer.java

```diff
@@ -204,7 +413,33 @@ public boolean canCheckEquality(JavaValue first, JavaValue second, Context conte
                                             if (innerClassNode.superName.equals("java/lang/Thread")) {
                                                 MethodNode clinitMethod = classes.get(strCl).getClassNode().methods.stream().filter(mn -> mn.name.equals("<clinit>")).findFirst().orElse(null);
                                                 if (clinitMethod != null) {
+                                                	//We don't want to run anything else (removes everything above it)
+                                                	AbstractInsnNode firstVaildCode = null;
+                                                	for(int i = 0; i < clinitMethod.instructions.size(); i++)
+                                                	{
+                                                		AbstractInsnNode ain = clinitMethod.instructions.get(i);
+                                                		if(ain.getOpcode() == Opcodes.PUTSTATIC
+                                                			&& ((FieldInsnNode)ain).desc.contains("BigInteger")
```

Comment:
Let's use the full name when possible, to minimize possible abuse since it's easy to forget implementation details like this